bentoml/_internal/server/metrics/prometheus.py

```diff
@@ -83,41 +103,59 @@ def mark_process_dead(self) -> None:
             self.prometheus_client.multiprocess.mark_process_dead(self._pid)
 
     def start_http_server(self, port: int, addr: str = "") -> None:
-        self.prometheus_client.start_http_server(
+        self.prometheus_client.start_http_server(  # type: ignore
             port=port,
             addr=addr,
             registry=self.registry,
         )
 
-    def generate_latest(self):
+    def generate_latest(self) -> bytes:
         if self.multiproc:
             registry = self.prometheus_client.CollectorRegistry()
             self.prometheus_client.multiprocess.MultiProcessCollector(registry)
-            return self.prometheus_client.generate_latest(registry)
+            return t.cast(bytes, self.prometheus_client.generate_latest(registry))
         else:
             return self.prometheus_client.generate_latest()
 
     def get_metrics_report(
-        self,
-    ) -> t.List[t.Dict[str, t.Union[str, float]]]:
-        metrics_text = self.generate_latest().decode()
+        self, filter_: list[str] | None = None
+    ) -> list[dict[str, str | float]] | None:
+        metrics_text = self.generate_latest().decode("utf-8")
+
         if not metrics_text:
             return []
 
-        from prometheus_client.parser import text_string_to_metric_families
+        if filter_ is not None:
+            filter_ = self._exc_paths + filter_
+        else:
+            filter_ = self._exc_paths
+
+        from prometheus_client.parser import (
+            text_string_to_metric_families,  # type: ignore
+        )
+
+        metric: ext.Metric
+
+        def truncate(metrics: dict[str, str | float]) -> bool:
+            for m in filter_:
+                if re.match(m, str(metrics["endpoint"])) is not None:
+                    return False
+            return True
+
+        report_: list[dict[str, str | float]] = []
 
         for metric in text_string_to_metric_families(metrics_text):
             if (
                 metric.type == "counter"
                 and metric.name.startswith("BENTOML_")
                 and metric.name.endswith("_request")
             ):
-                return [
+                metric_ = [
                     {**sample.labels, "value": sample.value}
                     for sample in metric.samples
                 ]
-
-        return []
+                report_ += list(filter(truncate, metric_))
```

Comment:
it should just return here to stop looping through other metrics - we should have only one metrics matching the requirements above